### ISSUE TITLE
[Azure Management Monitor] Alert Rules list by resource group throws deserialization error

### ISSUE THREAD

**Issue by vishrutshah** [Azure/azure-sdk-for-python#970]
Sample Code Snippet:

```
from azure.monitor import InsightsClient
from azure.mgmt.monitor import InsightsManagementClient
from azure.common.credentials import ServicePrincipalCredentials
import datetime
from azure.mgmt.resource.resources import ResourceManagementClient

# Replace this with your subscription id
subscription_id = 'xxxxxxxx-xxx-xxxx-xxxx-xxxxxxxxx'

# See above for details on creating different types of AAD credentials
credentials = ServicePrincipalCredentials(
    client_id = 'ABCDEFAB-1234-ABCD-1234-ABCDEFABCDEF',
    secret = 'XXXXXXXXXXXXXXXXXXXXXXXX',
    tenant = 'ABCDEFAB-1234-ABCD-1234-ABCDEFABCDEF'
)

mgmt_client = InsightsManagementClient(
    credentials,
    subscription_id
)
resource_client = ResourceManagementClient(
    credentials,
    subscription_id
)
resource_client.providers.register('Microsoft.Insights')

# Management Plane - Alert Rules
alertRules = mgmt_client.alert_rules.list_by_resource_group('vishrutrg')

for rule in alertRules:
    # assert isinstance(event, azure.monitor.models.EventData)
    print(rule)
```

Error:
```
Unable to deserialize {'odatatype': 'Microsoft.Azure.Management.Insights.Models.ThresholdRuleCondition'} into model <class 'azure.mgmt.monitor.models.rule_condition.RuleCondition'>. __init__() got an unexpected keyword argument 'odatatype'
```

**CC:**
@lmazuel

**Comment by lmazuel** [comment]
Autorest issue: https://github.com/Azure/autorest/issues/1729

**Comment by lmazuel** [comment]
Fixed by msrest 0.4.5
https://pypi.python.org/pypi/msrest/0.4.5
